packages/mjml-carousel/src/Carousel.js

```diff
@@ -0,0 +1,326 @@
+import { BodyComponent } from 'mjml-core'
+import range from 'lodash/range'
+import repeat from 'lodash/repeat'
+import min from 'lodash/min'
+import map from 'lodash/map'
+import crypto from 'crypto'
+
+import { msoConditionalTag } from 'mjml-core/lib/helpers/conditionalTag'
+
+export default class MjCarousel extends BodyComponent {
+
+  static allowedAttributes = {
+    'align': 'enum(left,center,right)',
+    'border-radius': 'unit(px,%)',
+    'icon-width': 'unit(px,%)',
+    'left-icon': 'string',
+    'padding': 'unit(px,%){1,4}',
+    'padding-top': 'unit(px,%)',
+    'padding-bottom': 'unit(px,%)',
+    'padding-left': 'unit(px,%)',
+    'padding-right': 'unit(px,%)',
+    'right-icon': 'string',
+    'thumbnails': 'enum(visible,hidden)',
+    'tb-border': 'string',
+    'tb-border-radius': 'unit(px,%)',
+    'tb-hover-border-color': 'color',
+    'tb-selected-border-color': 'color',
+    'tb-width': 'unit(px,%)'
+  }
+
+  static defaultAttributes = {
+    'align': 'center',
+    'border-radius': '6px',
+    'icon-width': '44px',
+    'left-icon': 'http://i.imgur.com/xTh3hln.png',
+    'right-icon': 'http://i.imgur.com/os7o9kz.png',
+    'thumbnails': 'visible',
+    'tb-border': '2px solid transparent',
+    'tb-border-radius': '6px',
+    'tb-hover-border-color': '#fead0d',
+    'tb-selected-border-color': '#ccc'
+  }
+
+  constructor(initialDatas = {}) {
+    super(initialDatas)
+    this.carouselId = crypto.randomBytes(6).toString('hex')
+  }
+
+  headStyle() {
+    const length = this.props.children.length
+    const { carouselId } = this
+
+    if (!length) return ''
+
+    const carouselCss = `
+    .mj-carousel {
+      -webkit-user-select: none;
+      -moz-user-select: none;
+      user-select: none;
+    }
+
+    .mj-carousel-icons-cell {
```

Comment:
You don't need to prefix it with the carouselID too ?

Reply:
No, this class is not involved with the checked/hover behaviors so it's fine

Reply:
Ok but cell width is related to current carousel no (cf the getAttribute right after) ?

Reply:
That's actually very true, i'll fix that